libraries/core_libs/consensus/src/dag/dag.cpp

```diff
@@ -475,29 +475,33 @@ void DagManager::getGhostPath(std::vector<blk_hash_t> &ghost) const {
 }
 
 // return {period, block order}, for pbft-pivot-blk proposing
-std::pair<uint64_t, std::vector<blk_hash_t>> DagManager::getDagBlockOrder(blk_hash_t const &anchor) {
+std::vector<blk_hash_t> DagManager::getDagBlockOrder(blk_hash_t const &anchor, uint64_t period) {
   SharedLock lock(mutex_);
-  // TODO: need to check if the anchor already processed
-  // if the period already processed
   std::vector<blk_hash_t> blk_orders;
 
+  if (period != period_ + 1) {
+    LOG(log_er_) << "getDagBlockOrder called with period " << period << ". Expected period " << period_ + 1
```

Comment:
I think change it to log_wr_?

Reply:
Like in the case happened in Testnet. There are at least have one node get enough cert votes and pushed period 5104 into PBFT chain. Then the node will be able to proposed block period at 5105. But for some other nodes who have not had period 5104, will fail at here. So those nodes will not cert vote for the block period 5105 until they push period 5104 into chain. That's not error and I think that will happen often